clang/test/Sema/debug-93066.cpp

```diff
@@ -0,0 +1,49 @@
+// RUN: %clang_cc1 -std=c++23 -fsyntax-only -verify %s
+
+struct S {
+  void f() {
+    ++this; // expected-error {{expression is not assignable}}
+    // expected-note@-1 {{add '*' to dereference it}}
```

Comment:
This is a bad diagnostic change: adding the `*` doesn't help.

Reply:
That’s true. Looking at this again, I’m candidly a bit confused that this seemed fine to me at the time...